core/src/blocktree.rs

```diff
@@ -356,13 +371,40 @@ impl Blocktree {
             write_batch.put::<cf::ErasureMeta>((slot, set_index), &erasure_meta)?;
         }
 
+        batch_processor.write(write_batch)?;
+
         if should_signal {
-            for signal in self.new_blobs_signals.iter() {
+            for signal in &self.new_blobs_signals {
                 let _ = signal.try_send(true);
             }
         }
 
-        batch_processor.write(write_batch)?;
+        if self.completed_slots.len() > 0 && !newly_completed_slots.is_empty() {
```

Comment:
similar question.  what does self.completed_slots.len() have to do with this clause?